src/aof.c

```diff
@@ -625,40 +593,24 @@ int aofUpgradePrepare(aofManifest *am) {
     am->curr_base_file_seq = 1;
     am->dirty = 1;
 
-    /* Persist the manifest file to temp AOF directory. */
-    sds amstr = getAofManifestAsString(am);
-    ret = writeAofManifestFile(tmp_dirname, amstr);
-    sdsfree(amstr);
-    if (ret == C_ERR) goto cleanup;
-    am->dirty = 0;
+    /* Persist the manifest file to AOF directory. */ 
+    if (persistAofManifest(am) != C_OK) {
+        exit(1);
+    }
 
-    /* Move the old AOF file to temp AOF directory. */
+    /* Move the old AOF file to AOF directory. */
+    sds aof_filepath = makePath(server.aof_dirname, server.aof_filename);
```

Comment:
No, we cannot add a suffix when rename here,  because this is the only way to identify the interrupted upgrade.

`am->base_aof_info->file_name` is `server.aof_filename` only appears when upgrading.

Otherwise, if a file with a suffix appears in the manifest but does not exist in the AOF dir, we consider this to be a serious error and the process exits directly.

Reply:
It's not the "only way" we identify it. it's at the very least a combination of a file in the manifest that's missing on the disk, and the presence of the old (legacy location) file on the disk. but i do agree that this make our interrupted upgrade detection better, so i'll drop it.

p.s. if we wanted we could have also added an indication in the manifest that this is the upgrade file.